From 9e53c6f3fa2c5ed3be4ac5a6877e3e344e867c07 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 9 May 2024 06:38:28 +0000
Subject: [PATCH] pcsc-lite 2.2.1 fix for cross build.

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 83ef6c3..a739b0f 100644
--- a/meson.build
+++ b/meson.build
@@ -224,7 +224,7 @@ install_headers(
 # data
 if get_option('polkit')
 install_data('doc/org.debian.pcsc-lite.policy',
-  install_dir : polkit_dep.get_variable('policydir'))
+  install_dir : polkit_dep.get_pkgconfig_variable('policydir', define_variable: [ 'datadir', get_option('prefix') + '/share' ]))
 endif
 
 # generate config.h
@@ -242,7 +242,7 @@ configure_file(output : 'pcscd.h',
 if get_option('libsystemd')
   systemd = dependency('systemd')
   unit = get_option('systemdunit')
-  systemdsystemunitdir = systemd.get_variable(pkgconfig : 'systemd' + unit + 'unitdir')
+  systemdsystemunitdir = systemd.get_pkgconfig_variable('systemd_user_unit_dir', define_variable: [ 'prefix', get_option('prefix') ])
 configure_file(output : 'pcscd.socket',
   input : 'etc/pcscd.socket.in',
   install_dir : systemdsystemunitdir,
-- 
2.31.1

