From 9bd8a180f3ebbb1e7621aa2b017922d228da732b Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 30 Nov 2023 14:11:47 +0000
Subject: [PATCH] set polkit_policy_dir to hard directory for cross build.

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 9a87c4d..b4d10b2 100755
--- a/configure
+++ b/configure
@@ -16486,7 +16486,7 @@ then :
 
 printf "%s\n" "#define HAVE_POLKIT 1" >>confdefs.h
 
-		 polkit_policy_dir=$($PKG_CONFIG polkit-gobject-1 --variable=policydir)
+		 polkit_policy_dir=/usr/share/polkit-1/actions
 		 POLICY_DIR=$polkit_policy_dir
 
 		 PCSCLITE_FEATURES="${PCSCLITE_FEATURES} polkit"
-- 
2.31.1

