From 298056cf51c6c470f34593cb2ff248da99796f8b Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 17 Jun 2024 08:22:24 +0000
Subject: [PATCH 08/11] Chromium 126 clang config add loongarch64.

---
 build/config/clang/BUILD.gn | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/config/clang/BUILD.gn b/build/config/clang/BUILD.gn
index 1a3a7e5440..a2a5d464e5 100644
--- a/build/config/clang/BUILD.gn
+++ b/build/config/clang/BUILD.gn
@@ -175,6 +175,8 @@ template("clang_lib") {
           _dir = "armv7-unknown-linux-gnueabihf"
         } else if (current_cpu == "arm64") {
           _dir = "aarch64-unknown-linux-gnu"
+        } else if (current_cpu == "loong64") {
+          _dir = "loongarch64-unknown-linux-gnu"
         } else {
           assert(false)  # Unhandled cpu type
         }
-- 
2.31.1

