From 8e369f80ce819ff4c73c5499b8051c7f807a144a Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 8 Aug 2024 01:27:28 +0000
Subject: [PATCH] qt6 6.7.2 fix for gcc 15.

---
 qtshadertools/src/3rdparty/glslang/SPIRV/SpvBuilder.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qtshadertools/src/3rdparty/glslang/SPIRV/SpvBuilder.h b/qtshadertools/src/3rdparty/glslang/SPIRV/SpvBuilder.h
index 79d2681..d3b4e0d 100644
--- a/qtshadertools/src/3rdparty/glslang/SPIRV/SpvBuilder.h
+++ b/qtshadertools/src/3rdparty/glslang/SPIRV/SpvBuilder.h
@@ -58,6 +58,7 @@ namespace spv {
 #include <algorithm>
 #include <map>
 #include <memory>
+#include <cstdint>
 #include <set>
 #include <sstream>
 #include <stack>
-- 
2.31.1

