From a2278e0ac94c39bc4f791206ec573c3b0bcc2f4d Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 13 May 2025 14:25:08 +0000
Subject: [PATCH 01/11] qt6 6.9.0 webengine configure add loongarch64 support.

---
 qtwebengine/configure.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qtwebengine/configure.cmake b/qtwebengine/configure.cmake
index e3961c2516..35c878f70c 100644
--- a/qtwebengine/configure.cmake
+++ b/qtwebengine/configure.cmake
@@ -302,7 +302,7 @@ qt_webengine_configure_check("supported-platform"
 )
 
 if(LINUX AND CMAKE_CROSSCOMPILING)
-    set(supported_targets "arm" "arm64" "armv7-a" "x86_64")
+    set(supported_targets "arm" "arm64" "armv7-a" "x86_64" "loongarch64")
     qt_webengine_configure_check("supported-arch"
         MODULES QtWebEngine QtPdf
         CONDITION TEST_architecture_arch IN_LIST supported_targets
-- 
2.31.1

