From 6b7431553b15b523b5e6bfc12a374c10397a4f69 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Fri, 15 Dec 2023 06:12:45 +0000
Subject: [PATCH] powerdevil 5.90 set None to suspend session by default.

---
 daemon/powerdevilsettingsdefaults.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/powerdevilsettingsdefaults.cpp b/daemon/powerdevilsettingsdefaults.cpp
index d34ccd7..8208af1 100644
--- a/daemon/powerdevilsettingsdefaults.cpp
+++ b/daemon/powerdevilsettingsdefaults.cpp
@@ -120,7 +120,7 @@ int ProfileDefaults::defaultAutoSuspendIdleTimeoutSec(const QString &profileGrou
 
 unsigned int ProfileDefaults::defaultAutoSuspendType()
 {
-    return qToUnderlying(PowerButtonAction::Sleep);
+    return qToUnderlying(PowerButtonAction::NoAction);
 }
 
 unsigned int ProfileDefaults::defaultPowerButtonAction(bool isMobile)
-- 
2.31.1

