From 809ba663e087f215810424f1b7e24151aadae2ab Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 6 May 2026 08:02:39 +0000
Subject: [PATCH] remove vc_uniscr_check function for paste selection.

---
 drivers/tty/vt/vt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/tty/vt/vt.c b/drivers/tty/vt/vt.c
index f8e01f946..0df6c7a1d 100644
--- a/drivers/tty/vt/vt.c
+++ b/drivers/tty/vt/vt.c
@@ -3135,7 +3135,7 @@ static int vc_process_ucs(struct vc_data *vc, int *c, int *tc)
 		 * required. This is one such case as we need to remember
 		 * which displayed characters are double-width.
 		 */
-		vc_uniscr_check(vc);
+//		vc_uniscr_check(vc);
 		return 2;
 	}
 
@@ -3211,7 +3211,7 @@ static int vc_get_glyph(struct vc_data *vc, int tc)
 	 * character meaning the simple screen buffer won't hold the original
 	 * information, whereas the Unicode screen buffer always does.
 	 */
-	vc_uniscr_check(vc);
+//	vc_uniscr_check(vc);
 
 	/* Try getting a simpler fallback character. */
 	tc = ucs_get_fallback(tc);
-- 
2.31.1

