From e12bb1bdcb9334aa21d8ab0a571fbffffce24c9a Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 12 Oct 2023 16:10:12 +0000
Subject: [PATCH] thunderbird 115.3.2 fix tools/power/mach_commands for
 python3.12

---
 tools/power/mach_commands.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/power/mach_commands.py b/tools/power/mach_commands.py
index 956dcdf..9d0dd66 100644
--- a/tools/power/mach_commands.py
+++ b/tools/power/mach_commands.py
@@ -2,6 +2,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import setuptools
 from distutils.version import StrictVersion
 
 from mach.decorators import Command, CommandArgument
-- 
2.31.1

