From d04a4b0e3148e4dba83719beab2b2c13d4069050 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 15 Jul 2025 05:11:02 +0000
Subject: [PATCH] set option_fam_jmp_cache force 0

---
 linux-user/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/linux-user/main.c b/linux-user/main.c
index dc303bc..64e1dbd 100644
--- a/linux-user/main.c
+++ b/linux-user/main.c
@@ -75,7 +75,7 @@ char real_exec_path[PATH_MAX];
 #ifdef CONFIG_LATA
 int indirect_jmp_opt_profile = 0;
 #ifdef CONFIG_LATA_INDIRECT_JMP
-int option_fam_jmp_cache = 1;
+int option_fam_jmp_cache = 0;
 #else
 int option_fam_jmp_cache = 0;
 #endif
-- 
2.31.1

