From 0d5a7065efbf10e603267433e1339f24780d396a Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 23 Nov 2023 11:25:52 +0000
Subject: [PATCH] libvorbisidec 1.2.1 fix configure error.

---
 configure.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.in b/configure.in
index e7f5690..28d292a 100644
--- a/configure.in
+++ b/configure.in
@@ -119,7 +119,6 @@ fi
 if test "x$HAVE_OGG" = "xno"
 then
   dnl fall back to the old school test
-  XIPH_PATH_OGG(, AC_MSG_ERROR(must have Ogg installed!))
   libs_save=$LIBS
   LIBS="$OGG_LIBS"
   AC_CHECK_FUNC(oggpack_writealign, , AC_MSG_ERROR(Ogg >= 1.0 required !))
-- 
2.31.1

