From ab5bda0085bf23df08d07710a63a982ae8963330 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sun, 27 Oct 2024 05:28:53 +0000
Subject: [PATCH 01/15] qt6 webengine configure add loongarch64 support.

---
 qtwebengine/configure.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qtwebengine/configure.cmake b/qtwebengine/configure.cmake
index 81b1c95bbe..789cde6074 100644
--- a/qtwebengine/configure.cmake
+++ b/qtwebengine/configure.cmake
@@ -479,7 +479,7 @@ add_check_for_support(
    MESSAGE "Build can be done only on Linux, Windows, macO, iOS and Android."
 )
 if(LINUX AND CMAKE_CROSSCOMPILING)
-   set(supportedTargets "arm" "arm64" "armv7-a" "x86_64")
+   set(supportedTargets "arm" "arm64" "armv7-a" "x86_64" "loongarch64")
    add_check_for_support(
        MODULES QtWebEngine QtPdf
        CONDITION TEST_architecture_arch IN_LIST supportedTargets
-- 
2.31.1

