From 876879355d8bcc1d81673770aca2892c6aee1a6a Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 17 Jun 2024 08:16:53 +0000
Subject: [PATCH 05/11] Chromium 126 base system sys_info add loongarch64
 support.

from https://github.com/loongson/chromium/tree/loongarch-patches/chromium120/new-world reference.
---
 base/system/sys_info.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/base/system/sys_info.cc b/base/system/sys_info.cc
index 38bde5a139..68d8a2f460 100644
--- a/base/system/sys_info.cc
+++ b/base/system/sys_info.cc
@@ -257,6 +257,8 @@ std::string SysInfo::ProcessCPUArchitecture() {
   return "ARM_64";
 #elif defined(ARCH_CPU_RISCV64)
   return "RISCV_64";
+#elif defined(ARCH_CPU_LOONGARCH64)
+  return "LOONGARCH_64";
 #else
   return std::string();
 #endif
-- 
2.31.1

