From ba087b57562adf1089ac8b101f35fea9b0b76937 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sat, 16 Sep 2023 15:13:45 +0000
Subject: [PATCH] glean_parser remove precise_timestamps args.

---
 glean_parser/util.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/glean_parser/util.py b/glean_parser/util.py
index edaeed9..1d1eec4 100644
--- a/glean_parser/util.py
+++ b/glean_parser/util.py
@@ -524,7 +524,6 @@ ping_args = [
     "name",
     "include_client_id",
     "send_if_empty",
-    "precise_timestamps",
     "reason_codes",
 ]
 
-- 
2.31.1

