From afac0c61f724443eaa3bf144d05b65ff9f128957 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 11 Jul 2023 12:15:14 +0000
Subject: [PATCH] fix build error at use without java.

---
 cui/source/options/optjava.cxx | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cui/source/options/optjava.cxx b/cui/source/options/optjava.cxx
index 17f128ce9..3bdafb323 100644
--- a/cui/source/options/optjava.cxx
+++ b/cui/source/options/optjava.cxx
@@ -937,6 +937,7 @@ OUString SvxJavaClassPathDlg::GetClassPath() const
 
 void SvxJavaClassPathDlg::SetClassPath( const OUString& _rPath )
 {
+#if HAVE_FEATURE_JAVA
     if ( m_sOldPath.isEmpty() )
         m_sOldPath = _rPath;
     m_xPathList->clear();
@@ -962,6 +963,9 @@ void SvxJavaClassPathDlg::SetClassPath( const OUString& _rPath )
         m_xPathList->select(0);
     }
     SelectHdl_Impl(*m_xPathList);
+#else
+    (void) this;
+#endif
 }
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
-- 
2.31.1

