From 2fd2207cc5a9d1c4bba01d499f14f26ceac63ab5 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sun, 14 Jan 2024 07:54:45 +0000
Subject: [PATCH] libreoffice 24.2.0 external curl configure add --without-ssl.

---
 external/curl/ExternalProject_curl.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/external/curl/ExternalProject_curl.mk b/external/curl/ExternalProject_curl.mk
index c63a8e046..fec80e6d1 100644
--- a/external/curl/ExternalProject_curl.mk
+++ b/external/curl/ExternalProject_curl.mk
@@ -37,7 +37,7 @@ $(call gb_ExternalProject_get_state_target,curl,build):
 	$(call gb_ExternalProject_run,build,\
 		$(gb_RUN_CONFIGURE) ./configure \
 			--without-amissl --without-bearssl --without-gnutls \
-			--without-mbedtls --without-rustls --without-wolfssl \
+			--without-mbedtls --without-rustls --without-wolfssl --without-ssl \
 			--disable-ftp --enable-http --enable-ipv6 \
 			--without-libidn2 --without-libpsl --without-librtmp \
 			--without-libssh2 --without-nghttp2 \
-- 
2.31.1

