From 8a0aea21f955e9c70733ff7f73a4596e7e7697c0 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 27 Feb 2024 03:06:37 +0000
Subject: [PATCH] libreoffice 24.2.1.1 fix for cross build.

---
 configure.ac | 6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index a773f80af..5deb0c847 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5757,7 +5757,7 @@ if test "$cross_compiling" = "yes"; then
     unset AR LD NM OBJDUMP PKG_CONFIG RANLIB READELF STRIP
     unset CPPUNIT_CFLAGS CPPUNIT_LIBS
     unset LIBXML_CFLAGS LIBXML_LIBS LIBXSLT_CFLAGS LIBXSLT_LIBS XSLTPROC
-    unset PKG_CONFIG_LIBDIR PKG_CONFIG_PATH
+    unset PKG_CONFIG_LIBDIR PKG_CONFIG_PATH PKG_CONFIG_SYSROOT_DIR
     if test -n "$CC_FOR_BUILD"; then
         export CC="$CC_FOR_BUILD"
         CC_BASE=`first_arg_basename "$CC"`
@@ -5767,6 +5767,7 @@ if test "$cross_compiling" = "yes"; then
         CXX_BASE=`first_arg_basename "$CXX"`
     fi
     test -n "$PKG_CONFIG_FOR_BUILD" && export PKG_CONFIG="$PKG_CONFIG_FOR_BUILD"
+    test -n "$LDFLAGS_FOR_BUILD" && export LDFLAGS="$LDFLAGS_FOR_BUILD"
     cd CONF-FOR-BUILD
 
     # Handle host configuration, which affects the cross-toolset too
@@ -5923,6 +5924,7 @@ if test "$cross_compiling" = "yes"; then
         CC
         CPPU_ENV
         CXX
+        LDFLAGS
         ILIB
         JAVA_HOME
         JAVAIFLAGS
@@ -6012,6 +6014,7 @@ else
     CC_FOR_BUILD="$CC"
     CPPU_ENV_FOR_BUILD="$CPPU_ENV"
     CXX_FOR_BUILD="$CXX"
+    LDFLAGS_FOR_BUILD="$LDFLAGS"
     INSTDIR_FOR_BUILD="$INSTDIR"
     INSTROOT_FOR_BUILD="$INSTROOT"
     LIBO_BIN_FOLDER_FOR_BUILD="$LIBO_BIN_FOLDER"
@@ -6032,6 +6035,7 @@ AC_SUBST(SDKDIRNAME_FOR_BUILD)
 AC_SUBST(WORKDIR_FOR_BUILD)
 AC_SUBST(CC_FOR_BUILD)
 AC_SUBST(CXX_FOR_BUILD)
+AC_SUBST(LDFLAGS_FOR_BUILD)
 AC_SUBST(CPPU_ENV_FOR_BUILD)
 
 dnl ===================================================================
-- 
2.31.1

