From 1e2d7fedaa7ba22750f14f8a688d18e13a1e892e Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Fri, 8 Sep 2023 08:47:43 +0000
Subject: [PATCH] libreoffice fix for gcc 14.

---
 store/source/storbase.hxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/store/source/storbase.hxx b/store/source/storbase.hxx
index feac0db96..3f6368cb8 100644
--- a/store/source/storbase.hxx
+++ b/store/source/storbase.hxx
@@ -34,6 +34,7 @@
 
 #include <memory>
 #include <utility>
+#include <algorithm>
 
 /** @file store common internals.
 */
-- 
2.31.1

