diff --git a/netwerk/dns/prepare_tlds.py b/netwerk/dns/prepare_tlds.py
index 53d0bf526d..86f5b0fb2b 100644
--- a/netwerk/dns/prepare_tlds.py
+++ b/netwerk/dns/prepare_tlds.py
@@ -4,7 +4,6 @@
 
 import codecs
 import encodings.idna
-import imp
 import os
 import re
 import sys
diff --git a/python/mach/mach/main.py b/python/mach/mach/main.py
index 9ab880341d..922addfb2a 100644
--- a/python/mach/mach/main.py
+++ b/python/mach/mach/main.py
@@ -8,9 +8,9 @@
 import argparse
 import codecs
 import errno
-import imp
 import logging
 import os
+import types
 import sys
 import traceback
 import uuid
@@ -35,6 +35,20 @@ from .sentry import NoopErrorReporter, register_sentry
 from .telemetry import create_telemetry_from_environment, report_invocation_metrics
 from .util import UserError, setenv
 
+
+import importlib.util
+import importlib.machinery
+def load_source(modname, filename):
+    loader = importlib.machinery.SourceFileLoader(modname, filename)
+    spec = importlib.util.spec_from_file_location(modname, filename, loader=loader)
+    module = importlib.util.module_from_spec(spec)
+    # The module is always executed and not cached in sys.modules.
+    # Uncomment the following line to cache the module.
+    # sys.modules[module.__name__] = module
+    loader.exec_module(module)
+    return module
+
+
 SUGGEST_MACH_BUSTED_TEMPLATE = r"""
 You can invoke ``./mach busted`` to check if this issue is already on file. If it
 isn't, please use ``./mach busted file %s`` to report it. If ``./mach busted`` is
@@ -267,13 +281,13 @@ To see more help for a specific command, run:
             # Ensure parent module is present otherwise we'll (likely) get
             # an error due to unknown parent.
             if "mach.commands" not in sys.modules:
-                mod = imp.new_module("mach.commands")
+                mod = types.ModuleType("mach.commands")
                 sys.modules["mach.commands"] = mod
 
             module_name = f"mach.commands.{uuid.uuid4().hex}"
 
         try:
-            imp.load_source(module_name, str(path))
+            load_source(module_name, str(path))
         except IOError as e:
             if e.errno != errno.ENOENT:
                 raise
diff --git a/python/mozbuild/mozbuild/nodeutil.py b/python/mozbuild/mozbuild/nodeutil.py
index 8ec724ab89..809b05a43c 100644
--- a/python/mozbuild/mozbuild/nodeutil.py
+++ b/python/mozbuild/mozbuild/nodeutil.py
@@ -5,6 +5,7 @@
 import os
 import platform
 import subprocess
+import setuptools
 from distutils.version import StrictVersion
 
 from mozboot.util import get_tools_dir
diff --git a/testing/mozbase/mozprofile/mozprofile/prefs.py b/testing/mozbase/mozprofile/mozprofile/prefs.py
index 6c3b1173c8..a5f8a6db7c 100644
--- a/testing/mozbase/mozprofile/mozprofile/prefs.py
+++ b/testing/mozbase/mozprofile/mozprofile/prefs.py
@@ -12,7 +12,7 @@ import tokenize
 import mozfile
 import six
 from six import StringIO, string_types
-from six.moves.configparser import SafeConfigParser as ConfigParser
+from six.moves.configparser import ConfigParser as ConfigParser
 
 if six.PY3:
 
