From 827db683b0eb61077abfc46c079904739b5d647b Mon Sep 17 00:00:00 2001
From: yala <zhaojunchao@loongson.cn>
Date: Mon, 30 Jan 2023 16:08:31 +0800
Subject: [PATCH 4/6] update Makefile

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b29ec55..2c1a6b6 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ include stdlib/StdlibModules
 
 CAMLC = $(BOOT_OCAMLC) $(BOOT_STDLIBFLAGS) -use-prims runtime/primitives
 CAMLOPT=$(OCAMLRUN) ./ocamlopt$(EXE) $(STDLIBFLAGS) -I otherlibs/dynlink
-ARCHES=amd64 arm64 power s390x riscv
+ARCHES=amd64 arm64 power s390x riscv loongarch64
 VPATH = utils parsing typing bytecomp file_formats lambda middle_end \
   middle_end/closure middle_end/flambda middle_end/flambda/base_types \
   asmcomp driver toplevel tools
-- 
2.31.1

