From 4409e98013c404eff54bca91d3fe7c4adef33799 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 1 Aug 2023 10:30:46 +0000
Subject: [PATCH] fix for glibc 2.38.

---
 alsactl/init_sysdeps.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/alsactl/init_sysdeps.c b/alsactl/init_sysdeps.c
index 3aca1b4..dc7fd41 100644
--- a/alsactl/init_sysdeps.c
+++ b/alsactl/init_sysdeps.c
@@ -17,7 +17,7 @@
  *
  */
 
-#if defined(__GLIBC__) && !(defined(__UCLIBC__) && defined(__USE_BSD))
+#if 0 //defined(__GLIBC__) && !(defined(__UCLIBC__) && defined(__USE_BSD))
 static size_t strlcpy(char *dst, const char *src, size_t size)
 {
 	size_t bytes = 0;
-- 
2.31.1

