From 396b743c5f7e862f1588798c01a69b9421431820 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 27 Aug 2024 03:32:42 +0000
Subject: [PATCH] libplasma 6.1.4 fix for build error.

---
 autotests/declarativetests/qmltest.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/autotests/declarativetests/qmltest.cpp b/autotests/declarativetests/qmltest.cpp
index b61cb79..349a543 100644
--- a/autotests/declarativetests/qmltest.cpp
+++ b/autotests/declarativetests/qmltest.cpp
@@ -3,6 +3,7 @@
 
 #include <QQmlEngine>
 #include <QtQuickTest>
+#include <quicktest.h>
 
 class Setup : public QObject
 {
-- 
2.31.1

