From a65fad3dacc7dd0da49cc7bcaf31a2aa50564a41 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 30 Oct 2024 08:59:10 +0000
Subject: [PATCH] siril add CMAKE_MAKE_PROGRAM to subprojects/htmesh for cross
 build.

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 98482e6..09df460 100644
--- a/meson.build
+++ b/meson.build
@@ -292,6 +292,7 @@ if not htmesh_dep.found()
 
   # Call CMake with `-DCMAKE_CXX_FLAGS="-I$(abs_srcdir)/htmesh"
   opt_var.add_cmake_defines({'CMAKE_CXX_FLAGS': '-I'+meson.current_source_dir()+'/subprojects/htmesh'})
+  opt_var.add_cmake_defines({'CMAKE_MAKE_PROGRAM': '@NINJA_COMMAND@'})
 if platform_windows
   opt_var.append_compile_args('cpp', '-Wno-unknown-pragmas')
 endif
-- 
2.31.1

