From d9a8a3a252b718f772e64372dcaa61ec88e24530 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sat, 29 Mar 2025 12:55:57 +0000
Subject: [PATCH] Fix build error for libxml2 2.14.

---
 storage/connect/plgxml.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/storage/connect/plgxml.h b/storage/connect/plgxml.h
index 82629e4..6c9fb6d 100644
--- a/storage/connect/plgxml.h
+++ b/storage/connect/plgxml.h
@@ -28,6 +28,8 @@ enum ElementType {               // libxml2
      XML_XINCLUDE_START     = 19,
      XML_XINCLUDE_END       = 20,
      XML_DOCB_DOCUMENT_NODE = 21};
+#else
+     #include <libxml/tree.h>
 #endif   // !BASE_BUFFER_SIZE
 
 //#if !defined(NODE_TYPE_LIST)
-- 
2.31.1

