From 2325024ea6a2131e81b901c36906351b915a21c3 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 7 Mar 2024 07:35:30 +0000
Subject: [PATCH] MaterialX 1.38.9 remove doc file to doc dir.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9cd5c85..c75b4dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -354,7 +354,7 @@ endif()
 # Install root-level documents
 if(NOT SKBUILD)
     install(FILES LICENSE CHANGELOG.md README.md THIRD-PARTY.md
-            DESTINATION .)
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/doc/MaterialX")
 
     set(MATERIALX_GEN_CONFIG_PATH "${MATERIALX_INSTALL_LIB_PATH}/cmake/${CMAKE_PROJECT_NAME}")
 
-- 
2.31.1

