From 147f12cfea42f41ca28d0ac466f33ddf489c86d4 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 27 Sep 2023 01:47:54 +0000
Subject: [PATCH] p11-kit fix systemd dir for cross build.

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 0f8c8da..4c70c7c 100644
--- a/meson.build
+++ b/meson.build
@@ -421,7 +421,7 @@ libsystemd_deps = []
 libsystemd = dependency('libsystemd', required: get_option('systemd'))
 systemd = dependency('systemd', required: get_option('systemd'))
 if libsystemd.found() and systemd.found()
-  systemduserunitdir = systemd.get_pkgconfig_variable('systemduserunitdir')
+  systemduserunitdir = systemd.get_pkgconfig_variable('systemd_user_unit_dir', define_variable: [ 'prefix', get_option('prefix') ])
   conf.set('WITH_SYSTEMD', 1)
   libsystemd_deps += libsystemd
   with_systemd = true
-- 
2.31.1

