From 9d2f4b63c4a73de3028964c79dfceb83eebb4c10 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 26 Sep 2023 09:43:28 +0000
Subject: [PATCH] ndctl fix systemd and udev dir for cross build.

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index ef001f8..9ee91a0 100644
--- a/meson.build
+++ b/meson.build
@@ -159,9 +159,9 @@ endif
 
 if get_option('systemd').enabled()
   systemd = dependency('systemd', required : true)
-  systemdunitdir = systemd.get_pkgconfig_variable('systemdsystemunitdir')
+  systemdunitdir = systemd.get_pkgconfig_variable('systemdsystemunitdir', define_variable: ['systemd_system_unit_dir', '/usr/lib/systemd/system'])
   udev = dependency('udev', required : true)
-  udevdir = udev.get_pkgconfig_variable('udevdir')
+  udevdir = udev.get_pkgconfig_variable('udevdir', define_variable: ['udev_dir', '/usr/lib/udev'])
   udevrulesdir = udevdir / 'rules.d'
 endif
 
-- 
2.31.1

