From d7705cb38552baa2858613cc9ebd311184b741b6 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sun, 22 Mar 2026 09:04:24 +0000
Subject: [PATCH] pipewire 1.6.2 fix for glibc 2.43.

---
 spa/plugins/alsa/acp/compat.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/spa/plugins/alsa/acp/compat.h b/spa/plugins/alsa/acp/compat.h
index f7592e1..09a17e5 100644
--- a/spa/plugins/alsa/acp/compat.h
+++ b/spa/plugins/alsa/acp/compat.h
@@ -434,7 +434,7 @@ static inline char *pa_path_get_filename(const char *p)
     char *fn;
     if (!p)
         return NULL;
-    if ((fn = strrchr(p, PA_PATH_SEP_CHAR)))
+    if ((fn = (char *)strrchr(p, PA_PATH_SEP_CHAR)))
         return fn+1;
     return (char*) p;
 }
@@ -471,7 +471,7 @@ static inline char *pa_replace(const char*s, const char*a, const char *b)
 	for (;;) {
 		const char *p;
 
-		if (!(p = strstr(s, a)))
+		if (!(p = (const char *)strstr(s, a)))
 			break;
 
 		pa_array_add_data(&res, s, p-s);
-- 
2.31.1

