From b608224f3b7275a80415ebce106f11ad410a3d0c Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 10 Sep 2024 12:05:40 +0000
Subject: [PATCH 2/2] blender BLI build config add loongarch64 define.

---
 source/blender/blenlib/BLI_build_config.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/source/blender/blenlib/BLI_build_config.h b/source/blender/blenlib/BLI_build_config.h
index 232c2fb..fe25a60 100644
--- a/source/blender/blenlib/BLI_build_config.h
+++ b/source/blender/blenlib/BLI_build_config.h
@@ -337,6 +337,11 @@
 #    define ARCH_CPU_32_BITS 1
 #    define ARCH_CPU_BIG_ENDIAN 1
 #  endif
+#elif defined(__loongarch_lp64)
+#  define ARCH_CPU_LOONGARCH_FAMILY 1
+#  define ARCH_CPU_LOONGARCH64 1
+#  define ARCH_CPU_64_BITS 1
+#  define ARCH_CPU_LITTLE_ENDIAN 1
 #else
 #  error Please add support for your architecture in BLI_build_config.h
 #endif
@@ -407,5 +412,8 @@
 #if !defined(ARCH_CPU_X86_64)
 #  define ARCH_CPU_X86_64 0
 #endif
+#if !defined(ARCH_CPU_LOONGARCH64)
+#  define ARCH_CPU_LOONGARCH64 0
+#endif
 
 /** \} */
-- 
2.31.1

