From 9da7e9985fcf27b727ad4515dec23cf3fb6b7e1f Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 13 Nov 2023 06:10:02 +0000
Subject: [PATCH] libdatachannel fix for gcc 14.

---
 src/impl/icetransport.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/impl/icetransport.cpp b/src/impl/icetransport.cpp
index 3a19e92..5d61ff4 100644
--- a/src/impl/icetransport.cpp
+++ b/src/impl/icetransport.cpp
@@ -15,6 +15,7 @@
 #include <iostream>
 #include <random>
 #include <sstream>
+#include <algorithm>
 
 #ifdef _WIN32
 #include <winsock2.h>
-- 
2.31.1

