From 2809f7308c8a524afd1ffaa31ccf6cba88e3fac2 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Fri, 8 Mar 2024 09:59:47 +0000
Subject: [PATCH] kdenlive fix opentimelineio name to OpenTimelineIO.

---
 src/pythoninterfaces/otioconvertions.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pythoninterfaces/otioconvertions.cpp b/src/pythoninterfaces/otioconvertions.cpp
index 6fa7f98..7e2164a 100644
--- a/src/pythoninterfaces/otioconvertions.cpp
+++ b/src/pythoninterfaces/otioconvertions.cpp
@@ -20,7 +20,7 @@
 OtioConvertions::OtioConvertions()
     : AbstractPythonInterface()
 {
-    addDependency(QStringLiteral("opentimelineio"), i18n("OpenTimelineIO conversion"));
+    addDependency(QStringLiteral("OpenTimelineIO"), i18n("OpenTimelineIO conversion"));
     addScript(QStringLiteral("otiointerface.py"));
     connect(this, &OtioConvertions::dependenciesAvailable, this, [&]() {
         if (QStandardPaths::findExecutable(QStringLiteral("otioconvert")).isEmpty()) {
-- 
2.31.1

