From c3e57b5867795af3e55ec0bc93fe89bd4c973e24 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 25 Sep 2023 08:34:48 +0000
Subject: [PATCH] libappimage fix for gcc 13.

---
 src/libappimage/utils/hashlib.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libappimage/utils/hashlib.h b/src/libappimage/utils/hashlib.h
index 089ee5c..070dded 100644
--- a/src/libappimage/utils/hashlib.h
+++ b/src/libappimage/utils/hashlib.h
@@ -2,6 +2,7 @@
 
 // system
 #include <istream>
+#include <cstdint>
 
 
 
-- 
2.31.1

