From d70971b17f7636d124680524ea67951fd736b5b6 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 17 Jun 2024 08:23:02 +0000
Subject: [PATCH 09/11] Chromium 131 rust config add loongarch64 support.

---
 build/config/compiler/BUILD.gn | 3 +++
 1 files changed, 3 insertions(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 6efe967eb0..63d42debe8 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1010,6 +1010,9 @@ config("compiler") {
     # directory from appearing in build outputs.
     "-Zremap-cwd-prefix=.",
   ]
+  if (current_cpu == "loong64") {
+    rustflags += [ "-Ccode-model=medium" ]
+  }
 
   if (!is_win || force_rustc_color_output) {
     # Colorize error output. The analogous flag is passed for clang. This must
