From 2be7281254db1f6eeffe2a5408dc3dee713a6a4d Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 31 Dec 2024 03:07:08 +0000
Subject: [PATCH] mpv 0.39 add qemu wrapper for cross build.

---
 TOOLS/gen-mpv-desktop.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/TOOLS/gen-mpv-desktop.py b/TOOLS/gen-mpv-desktop.py
index 7bbb33e..757d817 100755
--- a/TOOLS/gen-mpv-desktop.py
+++ b/TOOLS/gen-mpv-desktop.py
@@ -31,7 +31,7 @@ if __name__ == "__main__":
     if not mpv_desktop["X-KDE-Protocols"]:
         raise ValueError("Missing X-KDE-Protocols entry in mpv.desktop file")
 
-    mpv_protocols = check_output([sys.argv[2], "--no-config", "--list-protocols"], encoding="UTF-8")
+    mpv_protocols = check_output(["qemu-targetarch", sys.argv[2], "--no-config", "--list-protocols"], encoding="UTF-8")
     mpv_protocols = set(line.strip(" :/") for line in mpv_protocols.splitlines() if "://" in line)
     if len(mpv_protocols) == 0:
         raise ValueError("Unable to parse any protocols from mpv '--list-protocols'")
-- 
2.31.1

