From eb06f83014a238d1a3396c6996b59d1875fb203d Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 26 Jun 2023 11:56:28 +0000
Subject: [PATCH] opam remove create man file step for cross build.

---
 doc/man/dune | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/doc/man/dune b/doc/man/dune
index 7e38305..3d646c6 100644
--- a/doc/man/dune
+++ b/doc/man/dune
@@ -2,44 +2,6 @@
 ; assembled, if it was selected at configure-time (%{exe:opamMain.exe} is not executable in this
 ; case.
 
-(rule
-  (targets opam.1)
-  (deps opam-topics.inc opam-admin-topics.inc)
-  (action (with-stdout-to %{targets} (run %{bin:opam} --help=groff))))
-
-(install
-  (section man)
-  (package opam)
-  (files opam.1))
-
-(rule
-  (with-stdout-to opam-installer.1 (run %{bin:opam-installer} --help=groff)))
-
-(install
-  (section man)
-  (package opam-installer)
-  (files opam-installer.1))
-
-(executable
-  (name dune_man)
-  (libraries unix))
-
-(rule
-  (targets opam-topics.inc)
-  (deps %{bin:opam} using-built-opam)
-  (mode promote)
-  (action (with-stdout-to %{targets} (run %{exe:dune_man.exe} opam))))
-
-(rule
-  (targets opam-admin-topics.inc)
-  (deps %{bin:opam} using-built-opam)
-  (mode promote)
-  (action (with-stdout-to %{targets} (run %{exe:dune_man.exe} opam admin))))
-
-(include opam-topics.inc)
-
-(include opam-admin-topics.inc)
-
 ; This ensures that no opam command run will block asking for input
 (env
   (_ (env-vars ("OPAMYES" "no"))))
-- 
2.31.1

