From 1fceec131c8b2174ec3341bec43731dd549138c8 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Fri, 10 May 2024 08:59:15 +0000
Subject: [PATCH] dxvk fix for gcc 14.1.0

---
 src/d3d9/d3d9_mem.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/d3d9/d3d9_mem.cpp b/src/d3d9/d3d9_mem.cpp
index ec25624..c90d592 100644
--- a/src/d3d9/d3d9_mem.cpp
+++ b/src/d3d9/d3d9_mem.cpp
@@ -10,6 +10,7 @@
 #else
 #include <stdlib.h>
 #endif
+#include <algorithm>
 
 namespace dxvk {
 
-- 
2.31.1

