From f51cc8b9ae1779c151676bc9dd318914994513eb Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Fri, 29 Aug 2025 09:25:53 +0000
Subject: [PATCH 5/9] Chromium 139 base system sys_info add loongarch64
 support.

from https://github.com/loongson/chromium/tree/loongarch-patches/chromium120/new-world reference.
---
 base/system/sys_info.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/base/system/sys_info.cc b/base/system/sys_info.cc
index 000604bcb8..8b08fcc46c 100644
--- a/base/system/sys_info.cc
+++ b/base/system/sys_info.cc
@@ -256,6 +256,8 @@ std::string SysInfo::ProcessCPUArchitecture() {
   return "ARM_64";
 #elif defined(ARCH_CPU_RISCV64)
   return "RISCV_64";
+#elif defined(ARCH_CPU_LOONGARCH64)
+  return "LOONGARCH_64";
 #else
   return std::string();
 #endif
-- 
2.31.1

