From c1b5721a9de0b36735a89753e6037f1593b5f668 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Fri, 29 Aug 2025 09:29:43 +0000
Subject: [PATCH 7/9] Chromium 139 rust config add loongarch64 support.

---
 build/config/compiler/BUILD.gn | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 0c04c90368..b147d3026c 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1098,6 +1098,9 @@ config("compiler") {
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
   ]
+  if (current_cpu == "loong64") {
+    rustflags += [ "-Ccode-model=medium" ]
+  }
 
   if (!is_win || force_rustc_color_output) {
     # Colorize error output. The analogous flag is passed for clang. This must
-- 
2.31.1

