From 442d9e0268a11da9b20d8241372301b2c3285acb Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@loongson.cn>
Date: Wed, 3 Aug 2022 01:58:44 +0000
Subject: [PATCH] VLC fix for gcc13

---
 modules/demux/dash/mpd/TemplatedUri.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/demux/dash/mpd/TemplatedUri.hpp b/modules/demux/dash/mpd/TemplatedUri.hpp
index 1eeb70c..963cd11 100644
--- a/modules/demux/dash/mpd/TemplatedUri.hpp
+++ b/modules/demux/dash/mpd/TemplatedUri.hpp
@@ -22,6 +22,7 @@
 #define TEMPLATEDURI_HPP
 
 #include <string>
+#include <cstdint>
 
 namespace dash
 {
-- 
2.31.1

