From 6ce62ea331edac2f292a45428a0d76a0a0000be5 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 4 Sep 2024 08:48:07 +0000
Subject: [PATCH] librsvg 2.58.94 fix for cross build.

---
 gdk-pixbuf-loader/meson.build | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gdk-pixbuf-loader/meson.build b/gdk-pixbuf-loader/meson.build
index 5081844..d6dab8e 100644
--- a/gdk-pixbuf-loader/meson.build
+++ b/gdk-pixbuf-loader/meson.build
@@ -5,13 +5,15 @@ pixbufloader_sources = files(
 
 prefix = get_option('prefix')
 
+gdk_pixbuf_binarydir = pixbuf_dep.get_variable(pkgconfig: 'gdk_pixbuf_binarydir', pkgconfig_define: ['libdir', get_option('libdir')])
+
 pixbufloader = custom_target(
   'pixbufloader-svg',
   build_by_default: true,
   output: ['@0@pixbufloader_svg.@1@'.format(cargo_dylib_prefix, ext_dynamic)],
   console: true,
   install: true,
-  install_dir: pixbuf_dep.get_variable(pkgconfig: 'gdk_pixbuf_moduledir', pkgconfig_define: ['prefix', prefix]),
+  install_dir: pixbuf_dep.get_variable(pkgconfig: 'gdk_pixbuf_moduledir', pkgconfig_define: ['gdk_pixbuf_binarydir', gdk_pixbuf_binarydir]),
   depends: librsvg_rust_dep,
   depend_files: pixbufloader_sources,
   env: extra_env,
@@ -40,7 +42,7 @@ gdk_pixbuf_query_loaders = find_program(pixbuf_dep.get_variable(pkgconfig: 'gdk_
 
 pixbufloader_svg_install_args = [
   '--gdk-pixbuf-moduledir',
-  pixbuf_dep.get_variable(pkgconfig: 'gdk_pixbuf_moduledir', pkgconfig_define: ['prefix', prefix]),
+  pixbuf_dep.get_variable(pkgconfig: 'gdk_pixbuf_moduledir', pkgconfig_define: ['gdk_pixbuf_binarydir', gdk_pixbuf_binarydir]),
   pixbuf_dep.get_variable(pkgconfig: 'gdk_pixbuf_query_loaders', pkgconfig_define: ['prefix', prefix]),
   pixbuf_dep.get_variable(pkgconfig: 'gdk_pixbuf_cache_file', pkgconfig_define: ['prefix', prefix])
 ]
-- 
2.31.1

