From 40920759fd47d5c6f6b2d7eb32b5f26640932bc4 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 26 Jul 2023 07:29:32 +0000
Subject: [PATCH] gcc 13 add "-cmodule=medium" for loongarch libgcc.

---
 libgcc/config/loongarch/t-crtstuff | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgcc/config/loongarch/t-crtstuff b/libgcc/config/loongarch/t-crtstuff
index b8c36eb66..4412b52a6 100644
--- a/libgcc/config/loongarch/t-crtstuff
+++ b/libgcc/config/loongarch/t-crtstuff
@@ -2,4 +2,4 @@
 # We turn it off for crt*.o because it would make __EH_FRAME_BEGIN__ point
 # to .eh_frame data from crtbeginT.o instead of the user-defined object
 # during static linking.
-CRTSTUFF_T_CFLAGS += -fno-omit-frame-pointer -fno-asynchronous-unwind-tables
+CRTSTUFF_T_CFLAGS += -fno-omit-frame-pointer -fno-asynchronous-unwind-tables -mcmodel=medium
-- 
2.31.1

