From 15da9c869371fc1041618cdfd2f6291258a9f2ff Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 9 Sep 2024 14:45:33 +0000
Subject: [PATCH 1/3] chromium 128 add -mlsx and -mlasx for clang build.

---
 build/config/compiler/BUILD.gn | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index e5c0b29..a270b23 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1474,6 +1474,8 @@ config("compiler_cpu_abi") {
       cflags += [
         "-mabi=lp64d",
         "-mcmodel=medium",
+#        "-mlasx",
+        "-mlsx",
       ]
     } else if (current_cpu == "s390x") {
       cflags += [ "-m64" ]
-- 
2.31.1

