From 699317b45070d3a31cba4f45d8001ae60ff6912b Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 18 Mar 2025 01:54:56 +0000
Subject: [PATCH] cfitsio 4.6.0 fix lib to lib64.

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 90b628f..aaa5dca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,8 +291,8 @@ set_property(TARGET cfitsio APPEND PROPERTY
 )
 
 install(TARGETS cfitsio EXPORT cfitsioTargets
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib64
+  ARCHIVE DESTINATION lib64
   RUNTIME DESTINATION bin
   INCLUDES DESTINATION include
 )
@@ -316,7 +316,7 @@ configure_file(cmake/cfitsioConfig.cmake
   COPYONLY
 )
 
-set(ConfigPackageLocation lib/cmake/cfitsio)
+set(ConfigPackageLocation lib64/cmake/cfitsio)
 install(EXPORT cfitsioTargets
   FILE
     cfitsioTargets.cmake
@@ -337,7 +337,7 @@ install(
 
 # cfitsio.pc:
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cfitsio.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/cfitsio.pc @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cfitsio.pc DESTINATION lib/pkgconfig)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cfitsio.pc DESTINATION lib64/pkgconfig)
 
 #==============================================================================
 # Test programs:
-- 
2.31.1

