From 4931a4c0fbfe85c56711a5e70a90ced7dadf5c7f Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 10 Oct 2023 11:53:38 +0000
Subject: [PATCH] xwayland fix for Yongbao cross build.

---
 include/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/meson.build b/include/meson.build
index e6e3d55..d520f32 100644
--- a/include/meson.build
+++ b/include/meson.build
@@ -196,7 +196,7 @@ conf_data.set('DBE', '1')
 conf_data.set('DPMSExtension', build_dpms ? '1' : false)
 conf_data.set('DRI3', build_dri3 ? '1' : false)
 if build_glx
-    conf_data.set_quoted('DRI_DRIVER_PATH', dri_dep.get_pkgconfig_variable('dridriverdir'))
+    conf_data.set_quoted('DRI_DRIVER_PATH', dri_dep.get_pkgconfig_variable('dridriverdir', define_variable: [ 'dridriverdir', '/usr/' + get_option('libdir') + '/dri' ]))
 endif
 conf_data.set('HAS_SHM', build_mitshm ? '1' : false)
 conf_data.set('MITSHM', build_mitshm ? '1' : false)
-- 
2.31.1

