From c034ba59def6f6bfa348c99db1e4fe3072c8b178 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sun, 19 Oct 2025 03:09:24 +0000
Subject: [PATCH] html5lib 1.1 fix for python 14.

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index f84c128..41a6ab8 100644
--- a/setup.py
+++ b/setup.py
@@ -90,8 +90,8 @@ with open(join(here, "html5lib", "__init__.py"), "rb") as init_file:
         if (len(a.targets) == 1 and
                 isinstance(a.targets[0], ast.Name) and
                 a.targets[0].id == "__version__" and
-                isinstance(a.value, ast.Str)):
-            version = a.value.s
+                isinstance(a.value, ast.Constant)):
+            version = a.value.value
 
 setup(name='html5lib',
       version=version,
-- 
2.31.1

