From 076cc9a8d222099eda7d49215736279765c2ed5c Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 31 Dec 2024 12:06:00 +0000
Subject: [PATCH] OpenXR SDK 1.0.43 add loongarch64 support.

---
 src/common/platform_utils.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common/platform_utils.hpp b/src/common/platform_utils.hpp
index 59b5bf2..02dd1c3 100644
--- a/src/common/platform_utils.hpp
+++ b/src/common/platform_utils.hpp
@@ -71,8 +71,8 @@
 #define XR_ARCH_ABI "riscv64"
 #elif defined(__sparc__) && defined(__arch64__)
 #define XR_ARCH_ABI "sparc64"
-#elif defined(__loongarch64)
-#define XR_ARCH_ABI "loong64"
+#elif defined(__loongarch_lp64)
+#define XR_ARCH_ABI "loongarch64"
 #else
 #error "No architecture string known!"
 #endif
-- 
2.31.1

