From 4f5fb4cfed2f29f151f684a2edcb0efa7bcddfcf Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 7 Aug 2024 04:11:56 +0000
Subject: [PATCH] glslang 14.3.0 fix for gcc 15.

---
 SPIRV/SpvBuilder.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPIRV/SpvBuilder.h b/SPIRV/SpvBuilder.h
index 08057f4..454e2ef 100644
--- a/SPIRV/SpvBuilder.h
+++ b/SPIRV/SpvBuilder.h
@@ -58,6 +58,7 @@ namespace spv {
 #include <algorithm>
 #include <map>
 #include <memory>
+#include <cstdint>
 #include <set>
 #include <sstream>
 #include <stack>
-- 
2.31.1

