From 10a5e17e0b968a15d6522b6dd5ecd042740c857c Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 15 Jan 2025 13:31:44 +0000
Subject: [PATCH] lxml 5.3.0 fix __pyx_empty_tuple undefine.

---
 src/lxml/includes/etree_defs.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/lxml/includes/etree_defs.h b/src/lxml/includes/etree_defs.h
index 17d470d..0e1c39c 100644
--- a/src/lxml/includes/etree_defs.h
+++ b/src/lxml/includes/etree_defs.h
@@ -190,6 +190,8 @@ long _ftol2( double dblSource ) { return _ftol( dblSource ); }
   #define Py_TYPE(ob)   (((PyObject*)(ob))->ob_type)
 #endif
 
+PyObject *__pyx_empty_tuple = NULL;
+
 #define PY_NEW(T) \
      (((PyTypeObject*)(T))->tp_new( \
              (PyTypeObject*)(T), __pyx_empty_tuple, NULL))
-- 
2.31.1

