From c011e8e641942b7a71dba1969f473f1fb1500bf9 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 19 Jan 2026 11:46:56 +0000
Subject: [PATCH] qtquick3dphysics 6.10.1 PX_SIMD_DISABLED add loongarch64.

---
 .../PhysX/pxshared/include/foundation/PxPreprocessor.h        | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h b/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h
index 105e96f4e0..8ef3bfdff9 100644
--- a/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h
+++ b/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h
@@ -124,6 +124,8 @@ Architecture defines, see http://sourceforge.net/p/predef/wiki/Architectures/
 #define PX_PPC 1
 #elif defined(__mips__)
 #define PX_X64 1
+#elif defined(__loongarch_lp64)
+#define PX_X64 1
 #elif defined(__EMSCRIPTEN__)
 #define PX_WASM
     #if defined(__LP64__)
@@ -151,7 +153,7 @@ SIMD defines
 #endif
 
 /** Disable SIMD for webassembly, mips and arm64 */
-#if defined(__EMSCRIPTEN__) || defined(__mips__) || defined(_M_ARM64) || defined(_M_ARM)
+#if defined(__loongarch_lp64) || defined(__EMSCRIPTEN__) || defined(__mips__) || defined(_M_ARM64) || defined(_M_ARM)
 #define PX_SIMD_DISABLED 1
 #endif
 
-- 
2.31.1

