From 40315a01a5412f3778d9aa7ea195cde8b98a9763 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 24 Sep 2024 12:20:07 +0000
Subject: [PATCH] chromium 130 skia fix for loongarch64.

---
 skia/BUILD.gn | 1 +
 1 file changed, 1 insertions(+)

diff --git a/skia/BUILD.gn b/skia/BUILD.gn
index 669f9be..7fe4de5 100644
--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -83,6 +83,7 @@ config("skia_config") {
     "SK_DISABLE_LEGACY_GRAPHITE_IMAGES",
     "SK_DISABLE_LEGACY_DAWN_TEXTURE_INFO_FUNCS",
     "SK_DISABLE_LEGACY_DAWN_BACKEND_TEXTURE_FUNCS",
+    "SK_ENABLE_OPTIMIZE_SIZE",
   ]
 
   include_dirs = [
-- 
2.31.1

