From c80caeeac4d60b90eb93094b740105f04647e528 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sun, 12 Jan 2025 12:56:55 +0000
Subject: [PATCH] pyside6 fix compiler error for Qt 6.8.1

---
 sources/shiboken6/ApiExtractor/qtdocparser.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sources/shiboken6/ApiExtractor/qtdocparser.cpp b/sources/shiboken6/ApiExtractor/qtdocparser.cpp
index daafe68..19e8f15 100644
--- a/sources/shiboken6/ApiExtractor/qtdocparser.cpp
+++ b/sources/shiboken6/ApiExtractor/qtdocparser.cpp
@@ -74,7 +74,7 @@ QString QtDocParser::qdocModuleDir(const QString &pythonType)
     static const QHash<QString, QString> packageToModuleDir = getPackageToModuleDir();
 
     const QStringView package = packageFromPythonType(pythonType);
-    const auto it = packageToModuleDir.constFind(package);
+    const auto it = packageToModuleDir.constFind((const QString&)package);
     if (it == packageToModuleDir.cend()) {
         const QString known = packageToModuleDir.keys().join(", "_L1);
         qCWarning(lcShibokenDoc, "Type from unknown package: \"%s\" (known: %s).",
-- 
2.31.1

