From 138d5bca9ddbc595a05af39a9629fa276296e013 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 14 Sep 2023 14:40:48 +0000
Subject: [PATCH] asio fix compiler error for gcc14

---
 src/Makefile.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 73a30ac..29f0e59 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -280,7 +280,6 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 @STANDALONE_FALSE@EXAMPLES_CPP03 = examples/cpp03
 @HAVE_CXX11_TRUE@EXAMPLES_CPP11 = examples/cpp11
-@HAVE_CXX14_TRUE@EXAMPLES_CPP14 = examples/cpp14
 @HAVE_CXX17_TRUE@EXAMPLES_CPP17 = examples/cpp17
 @HAVE_CXX20_TRUE@EXAMPLES_CPP20 = examples/cpp20
 SUBDIRS = \
-- 
2.31.1

