From 60b36151ed3d0fd94821c6e34a9457a9d0b0c008 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 21 Feb 2024 10:17:10 +0000
Subject: [PATCH] VTK 9.3.0 ThirdParty double-conversion add loongarch support.

---
 .../vtkdoubleconversion/double-conversion/utils.h                | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h b/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
index 1bd2970..e847aef 100644
--- a/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
+++ b/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
@@ -98,6 +98,7 @@ int main(int argc, char** argv) {
 #if defined(_M_X64) || defined(__x86_64__) || \
     defined(__ARMEL__) || defined(__avr32__) || defined(_M_ARM) || defined(_M_ARM64) || \
     defined(__hppa__) || defined(__ia64__) || \
+    defined(__loongarch__) || \
     defined(__mips__) || \
     defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__) || \
     defined(_POWER) || defined(_ARCH_PPC) || defined(_ARCH_PPC64) || \
-- 
2.31.1

