From 7ce867258026eaed1c8b2296146ea25c0db6ba88 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 15 Jan 2025 03:57:18 +0000
Subject: [PATCH] sqlite 3.48.0 disable install to strip for cross build.

---
 main.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main.mk b/main.mk
index a0b08e5..1def7d1 100644
--- a/main.mk
+++ b/main.mk
@@ -2017,7 +2017,7 @@ sqlite3d$(T.exe):	shell.c $(LIBOBJS0)
 		$(LDFLAGS.libsqlite3) $(LDFLAGS.readline)
 
 install-shell-0: sqlite3$(T.exe) $(install-dir.bin)
-	$(INSTALL) -s sqlite3$(T.exe) "$(install-dir.bin)"
+	$(INSTALL) sqlite3$(T.exe) "$(install-dir.bin)"
 install-shell-1:
 install: install-shell-$(HAVE_WASI_SDK)
 
@@ -2031,7 +2031,7 @@ sqldiff$(T.exe): $(sqldiff.$(LINK_TOOLS_DYNAMICALLY).deps)
 	$(sqldiff.$(LINK_TOOLS_DYNAMICALLY).rules)
 
 install-diff: sqldiff$(T.exe) $(install-dir.bin)
-	$(INSTALL) -s sqldiff$(T.exe) "$(install-dir.bin)"
+	$(INSTALL) sqldiff$(T.exe) "$(install-dir.bin)"
 #install: install-diff
 
 dbhash$(T.exe):	$(TOP)/tool/dbhash.c sqlite3.o sqlite3.h
-- 
2.31.1

